File: src/main/java/com/bajdcc/LALR1/grammar/type/ConvertToString.java
package com.bajdcc.LALR1.grammar.type;

import com.bajdcc.util.lexer.token.Token;
import com.bajdcc.util.lexer.token.TokenType;

/**
 * 【类型转换】转换为字符串类型
 *
 * @author bajdcc
 */
public class ConvertToString implements ITokenConventer {

	@Override
	public Token convert(Token token) {
		token.setType(TokenType.STRING);
		token.setObj(String.valueOf(token.getObj()));
		return token;
	}
}

File: src/main/java/com/bajdcc/LALR1/grammar/type/ConvertToString.kt
package com.bajdcc.LALR1.grammar.type

import com.bajdcc.util.lexer.token.Token
import com.bajdcc.util.lexer.token.TokenType

/**
 * 【类型转换】转换为字符串类型
 *
 * @author bajdcc
 */
class ConvertToString : ITokenConventer {

    override fun convert(token: Token): Token {
        token.type = TokenType.STRING
        token.obj = token.obj.toString()
        return token
    }
}
